src/backend/distributed/commands/table.c

```diff
@@ -700,6 +700,27 @@ PostprocessAlterTableSchemaStmt(Node *node, const char *queryString)
 }
 
 
+/*
+ * PreprocessAlterTableAddPrimaryKey converts the ALTER TABLE ... ADD PRIMARY KEY ...
+ * into ALTER TABLE ... ADD CONSTRAINT <conname> PRIMARY KEY format and returns the DDLJob
+ * to run this command in the workers.
+ */
+static List *
+PreprocessAlterTableAddPrimaryKey(AlterTableStmt *alterTableStatement, Oid relationId)
+{
```

Comment:
Maybe you can add an assert here that mentions in which condition we can enter this function, i.e., assert that we are in CONSTR_PRIMARY and the conname is null or something like that.

Reply:
I added an error check rather than an assert in deparser as we spoke earlier in order to catch conname is NULL case.

Reply:
Usually in `Preprocess` functions we add an assert at the beginning that also shows what this preprocess function is used for. I think we can conform to that with this function as well.
You can still leave the error check in the deparser, that's a different thing.